src/Hooks.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {OwnableUpgradeable, IERC4626, ERC20} from "src/Common.sol";
+import {IVault} from "src/interface/IVault.sol";
+import {YnETHx} from "src/YnETHx.sol";
+import {IHooks} from "src/interface/IHooks.sol";
+
+contract Hooks is OwnableUpgradeable, IHooks {
+    // performance denominated in ether(i.e. 1e18 = 100%)
+    uint256 public performanceFee;
+    address public performanceFeeRecipient;
+    IVault public immutable VAULT;
+    uint256 public immutable VAULT_DECIMALS;
+
+    constructor(address vault_) {
+        VAULT = IVault(payable(vault_));
+        VAULT_DECIMALS = ERC20(address(VAULT)).decimals();
+    }
+
+    function initialize(address owner_, uint256 performanceFee_, address performanceFeeRecipient_)
+        external
+        initializer
+    {
+        __Ownable_init(owner_);
+        if (performanceFee_ > 1 ether) revert InvalidPerformanceFee();
+        performanceFee = performanceFee_;
+        performanceFeeRecipient = performanceFeeRecipient_;
+    }
+
+    function afterProcessAccounting(uint256 exchangeRateBefore, uint256 exchangeRateAfter, uint256 totalSupplyBefore)
```

Comment:
let's write unit tests for this specific function 

and also let's double down maing sure we test with:

the 6 decimals default asset  and the 18 decimal  shares vaults (example ynETHx)

the 6 decimals default asset and 18 decimals shares vault (example ynUSDx)

the 6 decimals default asset and 6 decimals shares vault (example Flex Strategy for ynRWAx)